### PR TITLE
feat: New media route

### PR DESCRIPTION
**Pull request by Joao-Castilho**
New media route

### CODE REVIEW

File: controllers/media.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+  const mediaSearch = await models.Media.findOne({
+    where: {
+      url: { [Op.like]: req.query.mediaUrl },
+      orgId: req.params.orgId,
+    },
+    transaction,
+  })
+
+  if (!mediaSearch) return res.status(HTTPStatus.NOT_FOUND).send({ errors: ['Image not found'] })
```
Comment (s1moe2) [inline controllers/media.js:96]: if media object not found, return 204 nevertheless?

File: controllers/media.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  const path = req.query.mediaUrl.substr(req.query.mediaUrl.indexOf('/media'), req.query.mediaUrl.length)
+
+  const storageClient = Storage.getStorageClient()
+  await storageClient.deleteFile(path)
+
+  await models.Media.destroy({
```
Comment (s1moe2) [inline controllers/media.js:103]: consider switching?

File: models/media.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const media = (sequelize, DataTypes) => {
+  const Media = sequelize.define('media', {
+    id: {
```
Comment (s1moe2) [inline models/media.js:3]: consider using file as PK

File: routes/media.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+/**
+ * @openapi
+ * /media/{appId}:
+ *   post:
+ *     summary: Upload app images/media
+ *     tags: [App]
+ *     security:
+ *       - cookieAuth: []
+ *     parameters:
+ *       - name: orgId
```
Comment (s1moe2) [inline routes/media.js:16]: orgId vs route4 defined appId

File: routes/media.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+/**
+ * @openapi
+ * /media/{orgId}:
+ *   delete:
+ *     summary: Delete selected media objects
+ *     tags: [App]
+ *     security:
+ *       - cookieAuth: []
+ *     parameters:
+ *       - name: OrgId
```
Comment (s1moe2) [inline routes/media.js:85]: Orgid vs orgId

**Review by s1moe2 — CHANGES_REQUESTED** [review]

File: controllers/media.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+const uploadMedia = async (req, res) => {
+  const transaction = await sequelize.transaction()
+
+  if (!req.formdata || !req.formdata.files) {
+    return res.status(HTTPStatus.BAD_REQUEST).send({ errors: ['no files uploaded'] })
+  }
+
+  const files = []
+  const badTypes = []
+  for (const key in req.formdata.files['']) {
```
[the remark this answers is not in the document]
Comment (Joao-Castilho) [reply controllers/media.js:21]: It must be

File: controllers/media.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  } catch (err) {
+    log.error(err, 'uploadMedia: failed to remove temporary files')
+    return res.status(HTTPStatus.INTERNAL_SERVER_ERROR).send('uploadMedia: failed to remove temporary files')
+  }
+
+  const response = {
+    savedImages: [],
+    errors: [],
+  }
+  const imageURls = []
```
Comment (s1moe2) [inline controllers/media.js:53]: not being read

File: controllers/media.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (s1moe2) [reply controllers/media.js:18]: Should iterate `.files` and uploaded files should be identified by a key in the form-data request

File: controllers/media.js [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  if (!response.errors.length) delete response.errors
+  return res.status(HTTPStatus.OK).send(response)
+}
+
+const deleteMedia = async (req, res) => {
+  const transaction = await sequelize.transaction()
+
+  const file = req.query.mediaUrl.substr(req.query.mediaUrl.indexOf('/media/') + 7, req.query.mediaUrl.length)
```
Comment (s1moe2) [inline controllers/media.js:87]: Repeated code. Consider moving into function?

File: controllers/media.js [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+  const mediaSearch = await models.Media.findByPk(file, { transaction })
```
Comment (s1moe2) [inline controllers/media.js:88]: Should be part of the transaction

File: controllers/media.js (hunk 1 above)
Comment (s1moe2) [reply controllers/media.js:90]: Should commit/rollback the transaction

File: migrations/20211001142037-create-media.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        allowNull: false,
+        primaryKey: true,
+      },
+      url: {
+        type: Sequelize.STRING,
+        allowNull: false,
+      },
+      org_id: {
+        type: Sequelize.STRING,
+        allowNull: false,
```
Comment (s1moe2) [inline migrations/20211001142037-create-media.js:15]: Missing FK `references` object property

File: models/media.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    orgId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  }, {
+    timestamps: true,
+    underscored: true,
+    freezeTableName: true,
+  })
+
```
Comment (s1moe2) [inline models/media.js:21]: Missing relation to organization

**Review by s1moe2 — CHANGES_REQUESTED** [review]

File: controllers/media.js [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+    where: {
+      file: file,
+      orgId: req.params.orgId,
+    },
+    transaction,
+  })
+  await transaction.commit()
+
+  const storageClient = Storage.getStorageClient()
+  await storageClient.deleteFile(path)
```
Comment (s1moe2) [inline controllers/media.js:113]: Shouldn't this happen before the model destroy (and commit), therefore preventing the model to be deleted if the image delete fails?

File: migrations/20211001142037-create-media.js (hunk 10 above)
Comment (s1moe2) [reply migrations/20211001142037-create-media.js:15]: This is not resolved. The created table has no FK.
imagem

**Review by s1moe2 — CHANGES_REQUESTED** [review]

**Review by s1moe2 — APPROVED** [review]

**Pull request merged**